### PR TITLE
add changelog

### PR DESCRIPTION
**Pull request by marsishandsome**
### What problem does this PR solve? 

add changelog

All notable changes to this project will be documented in this file.

- add release notes for old releases
- add release notes for TiSpark-2.1.2
- add release notes for TiSpark-2.2.0

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
+# TiSpark Changelog
+All notable changes to this project will be documented in this file.
+
+## [TiSpark 2.2.0]
+### New Features
+* Natively support writing data to TiDB using Spark Data Source API
```
Comment (zhexuany) [inline CHANGELOG.md]: I guess we should use TiKV here?

Comment (marsishandsome) [reply CHANGELOG.md]: fixed

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+* Fix MatchErrorException may occur when Unsigned BigInt contains in group by columns (#780)
+* Fix IndexOutOfBoundException when trying to get pd member (#788)
+
+## [TiSpark 2.0]
+### Features
+* Work with Spark 2.3
+* Support use $database statement
+* Support show databases statement
+* Support show tables statement
+* No need to use TiContext.mapTiDBDatabase, use $database.$table to identify a table instead
```
Comment (birdstorm) [inline CHANGELOG.md]:
```suggestion
* No need to use `TiContext.mapTiDBDatabase`, use `$database.$table` to identify a table instead
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+* Fix index scan on partition table (#735)
+* Fix KeyNotInRegion may occur when retrieving rows by handle (#755)
+* Fix encode value long max (#761)
+* Fix MatchErrorException may occur when Unsigned BigInt contains in group by columns (#780)
+* Fix IndexOutOfBoundException when trying to get pd member (#788)
+
+## [TiSpark 2.0]
+### Features
+* Work with Spark 2.3
+* Support use $database statement
```
Comment (birdstorm) [inline CHANGELOG.md]:
```suggestion
* Support use `$database statement`
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
+# TiSpark Changelog
+All notable changes to this project will be documented in this file.
+
+## [TiSpark 2.2.0]
+### New Features
+* Natively support writing data to TiKV using Spark Data Source API
+* Support select from partition table [#916](https://github.com/pingcap/tispark/pull/916)
+* Release one tispark jar instead of two [#933](https://github.com/pingcap/tispark/pull/933)
+* add spark version to tispark udf ti_version [#943](https://github.com/pingcap/tispark/pull/943)
```
Comment (birdstorm) [inline CHANGELOG.md]:
```suggestion
* Add spark version to tispark udf ti_version [#943](https://github.com/pingcap/tispark/pull/943)
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
+# TiSpark Changelog
+All notable changes to this project will be documented in this file.
+
+## [TiSpark 2.2.0]
+### New Features
+* Natively support writing data to TiKV using Spark Data Source API
+* Support select from partition table [#916](https://github.com/pingcap/tispark/pull/916)
+* Release one tispark jar instead of two [#933](https://github.com/pingcap/tispark/pull/933)
```
Comment (birdstorm) [inline CHANGELOG.md]: Should we add something like:

Support Spark 2.3/2.4

File: CHANGELOG.md (hunk 2 above)
Comment (marsishandsome) [reply CHANGELOG.md]: fixed

File: CHANGELOG.md (hunk 3 above)
Comment (marsishandsome) [reply CHANGELOG.md]: fixed

File: CHANGELOG.md (hunk 4 above)
Comment (marsishandsome) [reply CHANGELOG.md]: fixed

File: CHANGELOG.md (hunk 5 above)
Comment (marsishandsome) [reply CHANGELOG.md]: fixed

**Pull request merged**